File: 2015/day04_2.py
# pylint: disable=missing-module-docstring, missing-function-docstring


import hashlib
from typing import Any


def solution(content: str) -> Any:
    index = 1
    while True:
        hashed = hashlib.md5(bytes(content + str(index), "utf8")).hexdigest()
        if hashed.startswith("000000"):
            return index
        index += 1


def main() -> None:
    with open("day04.dat", "r", encoding="utf8") as file:
        print(solution(file.read().strip()))


if __name__ == "__main__":
    main()

File: 2015/day04-2.ts
import { encodeHex } from "@std/encoding/hex";
import { md5 } from "@takker/md5";

function solution(content: string) {
  let index = 1;
  while (true) {
    const hashed = md5(content + index);
    if (encodeHex(hashed).startsWith("000000")) {
      return index;
    }
    index += 1;
  }
}

Deno.readTextFile("day04.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
